### PR TITLE
update QAT and add export API

### PR DESCRIPTION
**Pull request by elad-c**
1. Add weight quantization in case there are weights not to be fine-tuned
2. Add export method to remove QuantizeWrappers from the QAT model after fine-tuning
3. Update docsrc with new API

### CODE REVIEW

File: model_compression_toolkit/qat/keras/quantization_facade.py [hunk 1]
```diff
@@ -148,16 +152,114 @@ def keras_quantization_aware_training_init(in_model: Model,
...
 
-        qat_model, user_info = model_builder(tg, fw_info=fw_info)
+        qat_model, user_info = model_builder(tg, fw_info=fw_info, fw_impl=fw_impl)
 
         user_info.mixed_precision_cfg = bit_widths_config
 
         return qat_model, user_info, QUANTIZATION_CONFIGS_DICT
 
+
+    def keras_quantization_aware_training_export(in_model: Model):
```
Comment (haihabi) [inline model_compression_toolkit/qat/keras/quantization_facade.py:162]: Rename the function from export to finalizing or something alone the line( this is true in several locations).

File: model_compression_toolkit/qat/keras/model_builder.py [hunk 2]
```diff
@@ -22,35 +23,84 @@
...
+def is_qat_applicaple(node: common.BaseNode):
+    """
+    A function for deciding if a layer should be fine-tuned
+    Args:
+        node (Model): Keras model to quantize.
+
+    Returns:
+        A boolean whether the layer is to be wrapped with a QuantizeWrapper
+    """
+    is_class_quantized = node.layer_class in [tf.keras.layers.Conv2D,
```
Comment (reuvenperetz) [inline model_compression_toolkit/qat/keras/model_builder.py:50]: Use node.type

File: model_compression_toolkit/qat/keras/model_builder.py [hunk 3]
```diff
@@ -22,35 +23,84 @@
...
 from model_compression_toolkit import get_target_platform_capabilities
 
 DEFAULT_KERAS_TPC = get_target_platform_capabilities(TENSORFLOW, DEFAULT_TP_MODEL)
 
 
+def is_qat_applicaple(node: common.BaseNode):
+    """
+    A function for deciding if a layer should be fine-tuned
+    Args:
+        node (Model): Keras model to quantize.
```
Comment (reuvenperetz) [inline model_compression_toolkit/qat/keras/model_builder.py:45]: Fix comment

File: model_compression_toolkit/qat/keras/model_builder.py [hunk 4]
```diff
@@ -22,35 +23,84 @@
...
+
+    Returns:
+        A boolean whether the layer is to be wrapped with a QuantizeWrapper
+    """
+    is_class_quantized = node.layer_class in [tf.keras.layers.Conv2D,
+                                              tf.keras.layers.DepthwiseConv2D,
+                                              tf.keras.layers.Conv2DTranspose,
+                                              tf.keras.layers.Dense]
+
+    is_weight_quantization = node.final_weights_quantization_cfg.enable_weights_quantization
```
Comment (reuvenperetz) [inline model_compression_toolkit/qat/keras/model_builder.py:55]: Please use node's method is_weights_quantization_enabled instead

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/core/common/defaultdict.py [hunk 5]
```diff
@@ -56,3 +56,10 @@ def get(self, key: Any) -> Any:
             if self.default_factory is not None:
                 return self.default_factory()
             return None
+
+    def keys(self):
+        """
+        Get keys od known_dict
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/common/defaultdict.py:62]: Typo

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

**Pull request merged**